### PR TITLE
[Reviewer: EM] Chronos GR Configuration options

### PR DESCRIPTION
**Pull request by richardwhiuk**
This adds parsing and unit tests for Chronos GR, and updated documentation.

The new unit tests pass, with full code coverage.

I've also tested live, with the following configuration:

```
[sites]
local_site=sitea
remote_site=siteb=siteb.cw-ngv.com:8000
remote_site=sietc=sitec.cw-ngv.com:8001
remote_site=d
```

resulting in the following logs:

```
05-08-2016 13:17:31.743 UTC Status globals.cpp:261: Cluster view ID: 288265571265413128
05-08-2016 13:17:31.743 UTC Status globals.cpp:285: Configured remote site: siteb=siteb.cw-ngv.com:8000
05-08-2016 13:17:31.743 UTC Status globals.cpp:285: Configured remote site: sietc=sitec.cw-ngv.com:8001
05-08-2016 13:17:31.743 UTC Error globals.cpp:281: Ignoring remote site: d - Site must include name and address separated by =
```

---

This change is [Reviewable](https://reviewable.io/reviews/metaswitch/chronos/280)

### CODE REVIEW

File: src/globals.cpp [hunk 1]
```diff
@@ -257,6 +264,31 @@ void Globals::update_config()
...
+  set_local_site_name(conf_map["sites.local_site"].as<std::string>());
+
+  std::vector<std::string> remote_site_list = conf_map["sites.remote_site"].as<std::vector<std::string>>();
+  std::map<std::string, std::string> remote_sites;
+
+  for (std::vector<std::string>::iterator it = remote_site_list.begin();
+                                          it != remote_site_list.end();
+                                          ++it)
+  {
+    std::size_t pos = it->find("=");
```
Comment (eleanor-merry) [inline src/globals.cpp]: Worth using https://github.com/Metaswitch/cpp-common/blob/master/include/utils.h#L249 for this check instead?

File: doc/configuration.md [hunk 2]
```diff
@@ -42,12 +47,23 @@ The cluster configuration file has the following format:
...
+The geographic redundancy configuration file has the following format:
+
+    [sites]
+    local_site = local-site-name         # The name of the local site
+    remote_site = site-b=site-b:8000     # The name of a remote site, and the address
+    remote_site = site-c=site-c:7000     # of the remote site. If the address doesn't
+    remote_site = site-d=site-d:5000     # include a port the bind-port will be used
+    remote_site = site-e=site-d          # Each site is listed in a separate entry
+
+To update the geographic redundancy configuration, make the desired changes in `/etc/chronos/chronos_gr.conf`, and reload Chronos (e.g. `service chronos reload`). This doesn't impact service.
```
Comment (eleanor-merry) [inline doc/configuration.md]: To update the geographic redundancy configuration, make the desired changes in `/etc/chronos/chronos_gr.conf` on one Chronos node in the site, and run `/usr/share/clearwater/clearwater-config-manager/scripts/upload_chronos_gr_config`. To check the state of the config being picked up across the Chronos processes within the site run `/usr/share/clearwater/clearwater-queue-manager/scripts/check_chronos_gr_queue_state`

File: doc/configuration.md [hunk 3]
```diff
@@ -42,12 +47,23 @@ The cluster configuration file has the following format:
...
 Details of how to set up the configuration for clustering is [here](https://github.com/Metaswitch/chronos/blob/dev/doc/clustering.md).
 
-To update the cluster configuration, make the desired changes in `etc/chronos/chronos_cluster.conf`, and reload Chronos (e.g. `service chronos reload`). This doesn't impact service. 
+To update the cluster configuration, make the desired changes in `/etc/chronos/chronos_cluster.conf`, and reload Chronos (e.g. `service chronos reload`). This doesn't impact service.
+
+The geographic redundancy configuration file has the following format:
+
+    [sites]
+    local_site = local-site-name         # The name of the local site
+    remote_site = site-b=site-b:8000     # The name of a remote site, and the address
```
Comment (eleanor-merry) [inline doc/configuration.md]: Can the site name and address be different in the example?

File: doc/configuration.md (hunk 2 above)
Comment (richardwhiuk) [inline doc/configuration.md]: Presumably these scripts are part of the etcd plugins, not Chronos? Should we instead be documenting the interface between the etcd plugins and Chronos here?

**Comment by eleanor-merry** [comment]
LGTM

**Comment by eleanor-merry** [comment]
Closing this PR for now

**Pull request closed without merging**